routing/router.go

```diff
@@ -1759,7 +1759,25 @@ func (r *ChannelRouter) SendToRoute(hash lntypes.Hash, route *route.Route) (
 		paymentHash: hash,
 	}
 
-	attempt, err := sh.launch(route)
+	attempt, err := sh.launch(rt)
+
+	// With SendToRoute, it can happen that the route exceeds protocol
```

Comment:
Can't we check all this _before_ the payment is actually "launched"? This would also save us from extra DB access and free list expansion.

Reply:
So we can just have a `Validate() error` method (or something like that), on `route.Route` itself.

Reply:
Sure, that's reasonable.